master/buildbot/changes/github.py

```diff
@@ -0,0 +1,262 @@
+#
+# Copyright Buildbot Team Members
+
+import os
+import time
+from datetime import datetime
+
+from twisted.internet import defer
+from twisted.internet import reactor
+from twisted.python import log
+from twisted.web import client
+
+from buildbot.changes import base
+from buildbot.util.state import StateMixin
+from buildbot.util import ascii2unicode
+from buildbot.util import datetime2epoch
+from buildbot.util import epoch2datetime
+from buildbot.util import httpclientservice
+
+HOSTED_BASE_URL = "https://api.github.com"
+
+
+class GitHubPullrequestPoller(base.PollingChangeSource, StateMixin):
+    name = "GitHubPullrequestPoller"
+
+    compare_attrs = ("owner", "repo", "token", "branch", "pollInterval",
+                     "useTimestamps", "category", "project", "pollAtLaunch")
+
+    db_class_name = 'GitHubPullrequestPoller'
+
+    @defer.inlineCallbacks
+    def reconfigService(self,
+                        owner,
+                        repo,
+                        branches=None,
+                        pollInterval=10 * 60,
+                        useTimestamps=True,
+                        category=None,
+                        baseURL=None,
+                        project='',
+                        pullrequest_filter=True,
+                        token=None,
+                        name=None,
+                        encoding='utf-8',
+                        pollAtLaunch=False,
+                        **kwargs):
+
+        yield base.PollingChangeSource.reconfigService(self, **kwargs)
+
+        if baseURL is None:
+            baseURL = HOSTED_BASE_URL
+        if baseURL.endswith('/'):
+            baseURL = baseURL[:-1]
+
+        http_headers = {'User-Agent': 'Buildbot'}
+        if not token is None:
+            http_headers.update({'Authorization': 'token ' + token})
+
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, baseURL, headers=http_headers)
+
+        if not branches:
+            branches = ['master']
+
+        self.token = token
+        self.owner = owner
+        self.repo = repo
+        self.branches = branches
+        self.project = project
+        self.encoding = encoding
+
+        if hasattr(pullrequest_filter, '__call__'):
+            self.pullrequest_filter = pullrequest_filter
+        else:
+            self.pullrequest_filter = (lambda _: pullrequest_filter)
+
+        self.lastChange = time.time()
+        self.lastPoll = time.time()
+        self.useTimestamps = useTimestamps
+        self.category = category if callable(category) else ascii2unicode(
+            category)
+        self.project = ascii2unicode(project)
+
+    def describe(self):
+        return "GitHubPullrequestPoller watching the "\
+            "GitHub repository %s/%s, branch: %s" % (
+                self.owner, self.repo, self.branch)
+
+    def _getPulls(self):
+        self.lastPoll = time.time()
+        log.msg("GitHubPullrequestPoller: polling "
```

Comment:
I dont think it is a good idea to have a log for each and every polling.

Please use buildbot.util.logger for new code